src/dataflow-types/src/client.rs

```diff
@@ -658,6 +667,120 @@ pub mod partitioned {
 
     use timely::progress::frontier::MutableAntichain;
 
+    /// Buffer for partial results for a `TAIL`.
+    /// This exists so we can consolidate rows and
+    /// sort them by timestamp before passing them to the consumer
+    /// (currently, coord.rs).
+    struct PendingTail {
+        /// `frontiers[i]` is an antichain representing the times at which we may
+        /// still receieve results from worker `i`.
+        frontiers: HashMap<usize, Antichain<Timestamp>>,
+        /// The results (possibly unsorted) which have not yet been delivered.
+        buffer: Vec<(Row, Timestamp, Diff)>,
+        /// The number of unique shard IDs expected.
+        parts: usize,
```

Comment:
This seems perhaps redundant with the `parts` field of `PartitionedClientState`.